junit5/src/main/java/sunstone/api/SunstoneResource.java

```diff
@@ -0,0 +1,53 @@
+package sunstone.api;
+
+import com.azure.resourcemanager.AzureResourceManager;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.s3.S3Client;
+import sunstone.api.inject.Hostname;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Supported resource injections:
+ * <ul>
+ *     <li>{@link OnlineManagementClient}: {@link SunstoneResource#of()} and {@link SunstoneResource#hint()} is required
+ *          <ul>
+ *             <li>
+ *                 {@link SunstoneResourceHint#AWS_EC2_INSTANCE}: {@link SunstoneResource#of()} name of the instance ({@code Name} tag)
+ *             </li>
+ *             <li>
+ *                 {@link SunstoneResourceHint#AZ_VM_INSTANCE}: {@link SunstoneResource#of()} name of the instance
```

Comment:
Would it be possible to do it so that you don't need the hint here? You either have an AWS configuration, or an Azure configuration - you should be able to decide if it's an AWS or Azure instance you're looking for.

Reply:
No. You can have resources of the same name in AWS, so it may not be possible to determine what kind of resource you ale looking for. I.e. hostname of a loadbalancer or VM...